src/test/bridges/convex/ConvexStakingBridgeE2E.t.sol

```diff
@@ -0,0 +1,243 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 Aztec.
+pragma solidity >=0.8.4;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import {AztecTypes} from "rollup-encoder/libraries/AztecTypes.sol";
+import {BridgeTestBase} from "./../../aztec/base/BridgeTestBase.sol";
+import {ErrorLib} from "../../../bridges/base/ErrorLib.sol";
+
+import {ConvexStakingBridge} from "../../../bridges/convex/ConvexStakingBridge.sol";
+import {IConvexBooster} from "../../../interfaces/convex/IConvexBooster.sol";
+
+contract ConvexStakingBridgeE2ETest is BridgeTestBase {
+    address private constant BOOSTER = 0xF403C135812408BFbE8713b5A23a04b3D48AAE31;
+    address private constant BENEFICIARY = address(777);
+
+    address private curveLpToken;
+    address private convexLpToken;
+    address private representingConvexToken;
+    address private rctImplementation;
+    address private rctClone;
+    address private staker;
+    address private gauge;
+    address private stash;
+    address private crvRewards;
+    // The reference to the convex staking bridge
+    ConvexStakingBridge private bridge;
+
+    uint16[] public invalidPids = [48]; // define invalid pids
+    // map of invalid pools, mapping(poolId)
+    mapping(uint16 => bool) public invalidPoolIds;
+
+    // map of already tested pools, mapping(poolId => bool)
+    mapping(uint16 => bool) public alreadyTestedPoolIds;
+
+    // To store the id of the convex staking bridge after being added
+    uint256 private bridgeId;
+
+    function setUp() public {
+        staker = IConvexBooster(BOOSTER).staker();
+        _setupInvalidPoolIds();
+
+        // labels
+        vm.label(address(ROLLUP_PROCESSOR), "Rollup Processor");
+        vm.label(address(this), "E2E Test Contract");
+        vm.label(msg.sender, "MSG sender");
+        vm.label(BOOSTER, "Booster");
+        vm.label(staker, "Staker Contract Address");
+        vm.label(BENEFICIARY, "Beneficiary");
+    }
+
+    /**
+     * @notice Tests staking and withdrawing by constructing bridgeCallData and passing it directly
+     * to RollupProcessor function that initializes the interaction with the bridge.
+     * @dev Tests a randomly selected pool from all available pools.
+     */
+    function testStakeWithdrawFlow(uint64 _depositAmount, uint16 _poolId) public {
```

Comment:
You should test that the accumulation of the rewards works. Are the CRV and CVX rewards distributed based on time? If yes, modifying block time should be sufficient for that.

Reply:
Well, it is going to be a bit trickier because CRV and CVX rewards are based on the stacked amount, performance of the pool (APY) and time. Also in some cases rewards can just be claimed but not exchanged for ether (because just too low amount of rewards was accumulated).
Also it was already tested in the unit tests and here I just check that the rewards claimed can be zero or greater.

Reply:
So I will most likely need to tweek it to make it work

Reply:
If too low an amount of rewards is to be exchanged then you should not do the exchange at all since it would be a waste of gas. I would first implement the rewards accounting and then I would test if it works as expected. You could do that by manually minting the rewards to the bridge.